### PR TITLE
[ELY-187] - API changes for role mapping.

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: src/main/java/org/wildfly/security/authz/RoleMapper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     *
+     * @param rolesToMap the roles in their raw form to apply mapping.
+     * @return
+     */
+    Set<String> mapRoles(Set<String> rolesToMap);
+
+    /**
+     * <p>A default implementation that does nothing but return the given roles.</p>
+     */
+    RoleMapper DEFAULT_ROLE_MAPPER = rolesToMap -> rolesToMap;
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/authz/RoleMapper.java]: A better term would be "identity role mapper".

Comment (pedroigor) [inline src/main/java/org/wildfly/security/authz/RoleMapper.java]: You mean rename `RoleMapper` to `IdentityRoleMapper` ?

File: src/main/java/org/wildfly/security/auth/login/SecurityIdentity.java [hunk 2]
```diff
@@ -169,4 +170,13 @@ public void runAs(Runnable action) {
             securityDomain.setCurrentSecurityIdentity(old);
         }
     }
+
+    /**
+     * <p>Get the roles associated with this identity.</p>
+     *
+     * @return the roles.
+     */
+    public Set<String> getRoles() {
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/auth/login/SecurityIdentity.java]: As discussed, we should have some realm info structure stored on the SI which would contain a reference to the realm and its mapper(s).

File: src/main/java/org/wildfly/security/auth/login/SecurityDomain.java [hunk 3]
```diff
@@ -54,14 +57,18 @@
     private final NameRewriter[] postRealmRewriters;
     private final boolean anonymousAllowed;
     private final ThreadLocal<SecurityIdentity> currentSecurityIdentity = new ThreadLocal<>();
+    private final RoleMapper roleMapper;
+    private final Map<String, RoleMapper> realmRoleMappers;
 
-    SecurityDomain(final Map<String, SecurityRealm> realmMap, final String defaultRealmName, final NameRewriter[] preRealmRewriters, final RealmMapper realmMapper, final NameRewriter[] postRealmRewriters) {
+    SecurityDomain(final Map<String, SecurityRealm> realmMap, final String defaultRealmName, final NameRewriter[] preRealmRewriters, final RealmMapper realmMapper, final NameRewriter[] postRealmRewriters, RoleMapper roleMapper, Map<String, RoleMapper> realmRoleMappers) {
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/auth/login/SecurityDomain.java]: As discussed, we should pass the builder directly in to the constructor.

File: src/main/java/org/wildfly/security/authz/RoleMapper.java (hunk 1 above)
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/authz/RoleMapper.java]: No, rename ``DEFAULT_ROLE_MAPPER`` to ``IDENTITY_ROLE_MAPPER`` or even ``IDENTITY``.

**Pull request merged**